### ISSUE TITLE
Make Ellipsoid Factor macro recordable

### ISSUE THREAD

**Issue by mdoube** [bonej-org/BoneJ2#235]
**Describe the bug**
Ellipsoid Factor (IJ2 style) is not macro recordable. No entry appears in the Macro Recorder when Ellipsoid Factor completes.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run the Macro Recorder
2. Run Ellipsoid Factor

**Expected behavior**
An Ellipsoid Factor command should appear in the Macro Recorder

**Screenshots**
![image](https://user-images.githubusercontent.com/87103/80576338-9a03dc00-8a37-11ea-899a-7c9da8193b3b.png)

**Additional context**
Reported on the forum https://forum.image.sc/t/using-ellipsoid-factor-in-a-macro/36975

It is possible to record `run("Ellipsoid Factor");` but only if you cancel the run from the dialog box. There should be a long string of variables listed in the recorded macro command.

Also, all the other recorded commands are noisy and should be hidden from the user.

Other IJ2 ('Modern') plugins can record, for example [`Anisotropy`](https://github.com/bonej-org/BoneJ2/blob/master/Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/AnisotropyWrapper.java):
```
run("Anisotropy", "inputimage=net.imagej.ImgPlus@274042d9 directions=2000 lines=10000 samplingincrement=1.73 recommendedmin=true printradii=true printeigens=true displaymilvectors=false")
```

**Comment by rimadoma** [comment]
I don't think this is a quick fix. Maybe a [hacky workaround](https://forum.image.sc/t/scijava-command-macro-recording/35056/14) is possible, but AFAIK ImageJ2 style commands are intended to be scripted with Python. I know it's a big ask to let go of macros, but my working assumption during the project was always that they were a thing of the past. And I assumed macro-recording would get replaced by something else as well. It was a mistake not to discuss this more though.

**Comment by mdoube** [comment]
> I don't think this is a quick fix.

Is there not a pattern we can copy from `Anisotropy`, which is also a pure IJ2 plugin and is perfectly recorded by the macro recorder?

**Comment by mdoube** [comment]
> ImageJ2 style commands are intended to be scripted with Python

In that case it would for us to give users a hint in the docs about what commands to include. The beauty of the macro recorder is that users don't have to do much and the code is written for them, which is a nice bump to get started with simple procedural programming.

**Comment by rimadoma** [comment]
> Is there not a pattern we can copy from Anisotropy, which is also a pure IJ2 plugin and is perfectly recorded by the macro recorder?

That's good news then! I pessimistically assumed that this affected all the commands in _Modern_.

**Comment by rimadoma** [comment]
> In that case it would for us to give users a hint in the docs about what commands to include. The beauty of the macro recorder is that users don't have to do much and the code is written for them, which is a nice bump to get started with simple procedural programming.

Bottom line is I know little about macros / scripting in ImageJ. If macro recording does work, then that's great, but we should investigate how the other commands fare. @mdoube  Since you know most about macros, could you do that at some point? If there are problems, we'll try to fix them like this one. If not, we could at least add macro snippets to documentation that users could add to their otherwise recorded macros.

**Comment by mdoube** [comment]
Fixed by #238
